client/src/actionsAndLoaders/loaders.ts

```diff
@@ -107,10 +107,11 @@ export const tagLoader = async ({ params }: LoaderFunctionArgs) => {
 export const searchLoader = async ({ request, params }: LoaderFunctionArgs) => {
     const url = new URL(request.url)
     const q = url.searchParams.get('q') as string
-    const results = (await fetch(`${api}tasks-titles`)) as TypedResponse<Task[]>
-    const tasks = await selectSearchResult(results, q);
-    const selectedId = params.id;
-
+    const results = (await fetch(`${api}tasks-titles?title=${q}`)) as TypedResponse<Task[]>
+    if (!results.ok) throw new Error('Something went wrong!')
```

Comment:
feels like this is check should be part of the fetching. Perhaps we can create a helper